thriftclient/example_client_test.go

```diff
@@ -0,0 +1,29 @@
+package thriftclient_test
+
+import (
+	"context"
+
+	"github.com/apache/thrift/lib/go/thrift"
+
+	"github.com/reddit/baseplate.go/internal/gen-go/reddit/baseplate"
+	"github.com/reddit/baseplate.go/thriftclient"
+	"github.com/reddit/baseplate.go/tracing"
+)
+
+func ExampleMonitoredClient() {
+	// variables should be properly initialized in production code
+	var (
+		transport thrift.TTransport
+		factory   thrift.TProtocolFactory
+		tracer    *tracing.Tracer
+	)
+	// Create an actual service client
+	client := baseplate.NewBaseplateServiceClient(
+		// Use MonitoredClient to wrap a standard thrift client
+		thriftclient.NewMonitoredClientFromFactory(transport, factory),
+	)
+	// Create a context with a server span
+	ctx, _ := tracing.CreateServerSpanForContext(context.Background(), tracer, "test")
+	// Calls should be automatically wrapped using client spans
+	client.IsHealthy(ctx)
```

Comment:
I've never written Example code, would it make sense to check the error here to display that the return value should be checked?